tests/suites/test_suite_md.function

```diff
@@ -1,5 +1,4 @@
 /* BEGIN_HEADER */
-#include "mbedtls/md.h"
```

Comment:
This seems wrong. This file is going to test MD functions so surely it should include `md.h`?

I guess things still work because it's getting included by some other header, but we shouldn't rely on that. Files that call functions declared in `md.h` or use macros defined there, should `#include "md.h"` directly instead of relying on another header to do that for them.

This is the one example when I know it's wrong without having to dig, but I suspect it's not the only one. Please check the other instances as well.

We should only remove the include if the only thing use in this file are the `MD_CAN` macros. (So, removing the include from `ssl_ciphersuite.h` is probably right, but removing it from `rsa.c` is probably wrong.)

Reply:
> Note: `rsa.c` includes `md.h` in `rsa.h`

Good point. I even thought about it at some point, then evidently forgot when picking an example for "probably wrong" ^^